docstore/mongodocstore/codec.go

```diff
@@ -236,9 +237,9 @@ func (d decoder) MapLen() (int, bool) {
 	return 0, false
 }
 
-func (d decoder) DecodeMap(f func(key string, d2 driver.Decoder) bool) {
+func (d decoder) DecodeMap(f func(key string, d2 driver.Decoder, exactMatch bool) bool) {
 	for k, v := range d.val.(map[string]interface{}) {
-		if !f(k, decoder{v}) {
+		if !f(k, decoder{v, d.lowercaseFields}, !d.lowercaseFields) {
```

Comment:
It's strange that this ignores the exactMatch arg. Isn't this the only reason to have the arg?

Reply:
Same point.